src/components/CommandBar/CommandsList/index.tsx

```diff
@@ -0,0 +1,156 @@
+    (navigationItem: SearchNavigationResult) => {
+      router
+        .push(resolveUrlBySearchNavigationType(navigationItem.resultType, navigationItem.key))
+        .then(() => {
+          dispatch({ type: addRecentNavigation.type, payload: navigationItem });
+          dispatch({ type: setIsOpen.type, payload: false });
+        });
+    },
+    [dispatch, router],
+  );
+  useHotkeys(
+    'up',
+    onUpKeyClicked,
+    {
+      enabled: numberOfItems && selectedItemIndex !== 0,
+      enableOnTags: ['INPUT'],
+    },
+    [scrollToSelectedItem],
+  );
+  useHotkeys(
+    'down',
+    onDownKeyClicked,
+    {
+      enabled: numberOfItems && selectedItemIndex !== numberOfItems - 1,
+      enableOnTags: ['INPUT'],
+    },
+    [scrollToSelectedItem],
+  );
+  useHotkeys(
+    'enter',
+    () => {
+      navigateToItem(navigationItems[selectedItemIndex]);
+    },
+    { enabled: selectedItemIndex !== null, enableOnTags: ['INPUT'] },
+    [navigateToItem, navigationItems, selectedItemIndex],
+  );
+  const onRemoveItemClicked = (event: MouseEvent<Element>, navigationItemKey: number | string) => {
+    // to not allow the event to bubble up to the parent container
+    event.stopPropagation();
+    dispatch({ type: removeRecentNavigation.type, payload: navigationItemKey });
+  };
+
+  if (!numberOfItems) {
+    return (
+      <p className={styles.noResult}>
+        {isPreInput ? 'No recent navigations' : 'No navigation results'}{' '}
+      </p>
+    );
+  }
+
+  const onItemHover = (itemIndex: number) => {
+    setSelectedItemIndex(itemIndex);
+  };
+
+  return (
+    <ul role="listbox">
+      <li role="presentation">
+        <div className={styles.groupHeader} id="navigation">
+          {isPreInput ? 'Recent navigations' : 'Navigations'}
+        </div>
+        <ul role="group" aria-labelledby="navigation">
+          {navigationItems.map((navigationItem, index) => (
+            <li
+              ref={selectedItemIndex === index ? selectedItemRef : null}
+              role="option"
+              aria-selected={selectedItemIndex === index}
+              key={navigationItem.key}
+              className={classNames(styles.commandItem, {
+                [styles.selected]: selectedItemIndex === index,
+              })}
+              onClick={() => navigateToItem(navigationItem)}
+              onMouseOver={() => onItemHover(index)}
+            >
+              <div className={styles.commandItemBody}>
+                <span className={styles.commandPrefix}>
+                  <NavigateIcon />
+                </span>
+                {navigationItem.name}
+              </div>
+              <div className={styles.keyboardInputContainer}>
+                {isPreInput ? (
+                  <Button
+                    variant={ButtonVariant.Ghost}
+                    size={ButtonSize.Small}
+                    onClick={(e) => onRemoveItemClicked(e, navigationItem.key)}
+                  >
+                    X
```

Comment:
haha, the everlasting hack. Let's use an x icon here. Muhajir recently added a small cross icon from radix ui